src/Xhgui/Controller/Run.php

```diff
@@ -309,4 +324,17 @@ public function callgraphDataDot()
         $response['Content-Type'] = 'application/json';
         return $response->body(json_encode($callgraph));
     }
+
+    public function import()
+    {
+        $request = $this->_app->request();
+
+        $data = json_decode($request->getBody(), true);
```

Comment:
There is no authentication, authorization or checksums on this data right now. Should there be?

Reply:
Yes, I realize that. It was a conscious choice, since there were none of these in the rest of the project as well.

Furthermore, I assumed that XHGui, being a tool that is primarily used during development (imho), would be deployed in a somewhat trusted environment, like a staging server, or a developers' workstation.